Logging: Add HTTP API to change the log level at runtime (#9357)
grafana/loki: docs/sources/api/_index.md

In microservices mode, the `/ready` endpoint is exposed by all components.

## Change log level at runtime

```
GET /log_level
POST /log_level
```

> `/log_level` returns the log level in case of `GET` and allows to change the log level at runtime in case of `POST`.

Reviewer: ```suggestion
`/log_level` a `GET` returns the current log level and a `POST` lets you change the log level of a Loki process at runtime. This can be useful for accessing debugging information during an incident. Caution should be used when running  at the `debug` log level, as this produces a large volume of data.
```

Author: @JStickler Thank you for the suggestion - I was struggling to write the description myself and this makes a lot more sense.

I accepted your suggestion already, but would it be better to remove the endpoint from the beginning? Or use a colon to separate the endpoint from the description. What would you prefer?

Reviewer: >would it be better to remove the endpoint from the beginning? Or use a colon to separate the endpoint from the description. 

@Indransh Looking over the rest of the topic, it looks like most of the entries have the endpoint at the beginning of the description, and do not have a colon to separate the endpoint from the description, so lets follow that convention so that things look consistent.

Author: Thanks, then lets keep it consistent. I pushed a small fix (remove extra space) and re-requested your review.

Resolving this thread since the original suggestion was accepted.

Params:

> - `log_level`: A valid log level that can be passed as URL param (`?log_level=<level>`) or as form value in case of `POST`. Valid levels: [debug, info, warn, error]

Reviewer: ```suggestion
- `log_level`: A valid log level that can be passed as a URL param (`?log_level=<level>`) or as a form value in case of `POST`. Valid levels: [debug, info, warn, error]
```

> `/log_level` a `GET` returns the current log level and a `POST` lets you change the log level of a Loki process at runtime. This can be useful for accessing debugging information during an incident. Caution should be used when running  at the `debug` log level, as this produces a large volume of data.

Author: ```suggestion
A `GET` returns the current log level and a `POST` lets you change the log level of a Loki process at runtime. This can be useful for accessing debugging information during an incident. Caution should be used when running at the `debug` log level, as this produces a large volume of data.
```

@JStickler I accepted your original suggestion. Would it be better to remove the endpoint from the beginning? Or use a hyphen or colon to separate the endpoint from the description. What would you prefer?
